InternalError("Throwable(Object(java/lang/InternalError)
  backtrace=java/lang/StackTraceElement[4]
  detailMessage=String(\"Internal error: No virtual method found for class\")
  cause=Object(class java/lang/InternalError)
  stackTrace=java/lang/StackTraceElement[0]
  depth=int(4)
  suppressedExceptions=Object(class java/util/Collections$EmptyList)
):
stdout: === Interface Exception Handling Test ===
Testing implementations that don't throw:
ThrowsImplementation.mayThrowChecked
ThrowsImplementation.mayThrowMultiple
ThrowsImplementation.mayThrowUnchecked
ThrowsImplementation.noThrowsClause
NoThrowsImplementation.mayThrowChecked - no exceptions
NoThrowsImplementation.mayThrowMultiple - no exceptions
NoThrowsImplementation.mayThrowUnchecked - no exceptions
NoThrowsImplementation.noThrowsClause - no exceptions

Testing implementations that do throw:
ThrowsImplementation.mayThrowChecked
Caught IOException from mayThrowChecked: Implementation IOException
ThrowsImplementation.mayThrowMultiple
Caught IOException from mayThrowMultiple: Implementation IOException from multiple
ThrowsImplementation.mayThrowUnchecked
Caught RuntimeException from mayThrowUnchecked: Implementation IllegalArgumentException
ThrowsImplementation.noThrowsClause
Caught RuntimeException from noThrowsClause: Unchecked exception from no-throws method

Testing exception hierarchy:
ThrowsImplementation.throwsParent
ThrowsImplementation.throwsChild
ThrowsImplementation.throwsParent
Caught from throwsParent: Exception - Parent exception
ThrowsImplementation.throwsChild
Caught from throwsChild: FileNotFoundException - Child exception - FileNotFoundException

Testing static method exceptions:
ThrowsInterface.staticWithThrows
Caught static method exception: Static method exception

Testing default method exceptions:
